src/Build/BackEnd/Components/SdkResolution/SdkResolverContext.cs

```diff
@@ -12,13 +16,45 @@ namespace Microsoft.Build.BackEnd.SdkResolution
     /// </summary>
     internal sealed class SdkResolverContext : SdkResolverContextBase
     {
-        public SdkResolverContext(Framework.SdkLogger logger, string projectFilePath, string solutionPath, Version msBuildVersion, bool interactive)
+        private readonly IDictionary<string, string> _globalProperties;
+        private readonly LoggingContext _loggingContext;
+
+        public SdkResolverContext(Framework.SdkLogger logger, string projectFilePath, string solutionPath, Version msBuildVersion, bool interactive, IDictionary<string, string> globalProperties, LoggingContext loggingContext)
         {
             Logger = logger;
             ProjectFilePath = projectFilePath;
             SolutionFilePath = solutionPath;
             MSBuildVersion = msBuildVersion;
             Interactive = interactive;
+            _globalProperties = globalProperties;
+            _loggingContext = loggingContext;
+        }
+
+        public override string GetGlobalPropertyValue(string name)
```

Comment:
Why expose these to the resolver? AFAIK resolvers didn't have access to global properties before. I'd vote on keeping in that way to minimize their complexity.

Reply:
Looking at the rippling changes required to pipe global properties, I'm more convinced we shouldn't expose them.

Reply:
@jeffkl ?

Reply:
This is so that users can override settings that the NuGet SDK resolver uses.  Right now it can only get its properties from `nuget.config`

https://github.com/microsoft/msbuild/issues/2914

Reply:
I'd separate the changes in two PRs then. Making sdk resolution depend on global properties also requires changing sdk resolver caching: https://github.com/maneely/msbuild/pull/1#discussion_r314034927